synapse/rest/client/room_batch.py

```diff
@@ -123,6 +123,19 @@ async def on_POST(
                     errcode=Codes.INVALID_PARAM,
                 )
 
+        # Make sure that the prev_event_ids exist and aren't outliers - ie, they are
+        # regular parts of the room DAG where we know the state.
+        non_outlier_prev_events = await self.store.have_events_in_timeline(
+            prev_event_ids_from_query
+        )
+        for prev_event_id in prev_event_ids_from_query:
+            if prev_event_id not in non_outlier_prev_events:
+                raise SynapseError(
+                    HTTPStatus.BAD_REQUEST,
+                    "prev_event %s does not exist, or is an outlier" % (prev_event_id,),
+                    errcode=Codes.INVALID_PARAM,
+                )
```

Comment:
We should add a test for this case, [`tests/rest/client/test_room_batch.py`](https://github.com/matrix-org/synapse/blob/1f32b90b0ff0f4f4f499f95e8150efe438388270/tests/rest/client/test_room_batch.py)